sdk/core/azure-core/src/main/java/com/azure/core/http/rest/RestProxy.java

```diff
@@ -275,6 +276,14 @@ private HttpRequest configRequest(final HttpRequest request, final SwaggerMethod
             }
 
             request.getHeaders().set("Content-Type", contentType);
+            if (bodyContentObject instanceof BinaryData) {
+                BinaryData binaryData = (BinaryData) bodyContentObject;
+                if (binaryData.getLength() != null) {
+                    request.setHeader("Content-Length", binaryData.getLength().toString());
+                }
+                request.setBody(binaryData.toFluxByteBuffer());
```

Comment:
For future reviewers of the code, it might be nice to comment on your understanding of this line, to the effect of whether or not it results in a memory copy at any point.